crates/ruma-common/src/receipt.rs

```diff
@@ -0,0 +1,14 @@
+//! Common types for receipts.
+
+use ruma_serde::StringEnum;
+
+/// The type of receipt.
+#[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, StringEnum)]
```

Comment:
`PartialOrd`, `Ord`, `PartialEq` and `Eq` should probably not be derived, I'd prefer if they were implemented in terms of `AsRef<str>`. We could probably add corresponding derive macros to `ruma-serde-macros`, or reuse the macro-rules macros from `ruma-identifiers`.

Reply:
What option do you prefer?